src/wmtk/io/EdgeMeshReader.hpp

```diff
@@ -0,0 +1,51 @@
+#include <Eigen/Core>
+#include <filesystem>
+#include <fstream>
+
+namespace wmtk {
+class Mesh;
+
+class EdgeMeshReader
+{
+public:
```

Comment:
Make sure that only those functions are `public` that need to be. The rest should be `private`.